### ISSUE TITLE
Dative bonds interfere with kekulization and the perception of aromaticity

### ISSUE THREAD

**Issue by greglandrum** [rdkit/rdkit#1703]
**Description:**

- RDKit Version: 2017.3 and on (really before, but it wasn't trivial to construct an example then)
- Platform: all

```
In [2]: m = Chem.MolFromSmiles('c1ccccn1->[Fe]')
[16:52:46] Can't kekulize mol.  Unkekulized atoms: 0 1 2 3 4


In [3]: m = Chem.MolFromSmiles('C1=CC=CC=N1->[Fe]')

In [4]: m.Debug()
Atoms:
	0 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 0 chi: 0
	1 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 0 chi: 0
	2 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 0 chi: 0
	3 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 0 chi: 0
	4 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 0 chi: 0
	5 7 N chg: 0  deg: 3 exp: 3 imp: 0 hyb: 3 arom?: 0 chi: 0
	6 26 Fe chg: 0  deg: 1 exp: 1 imp: 0 hyb: 4 arom?: 0 chi: 0
Bonds:
	0 0->1 order: 2 conj?: 1 aromatic?: 0
	1 1->2 order: 1 conj?: 1 aromatic?: 0
	2 2->3 order: 2 conj?: 1 aromatic?: 0
	3 3->4 order: 1 conj?: 1 aromatic?: 0
	4 4->5 order: 2 conj?: 1 aromatic?: 0
	5 5->6 order: 17 conj?: 0 aromatic?: 0
	6 5->0 order: 1 conj?: 1 aromatic?: 0
```

**Comment by greglandrum** [comment]
Zero-order bonds don't mess with aromaticity perception, but they do interfere with kekulization:
```
In [7]: m = Chem.MolFromSmiles('C1=CC=CC=N1.[Fe]',sanitize=False)

In [8]: m = Chem.RWMol(m)

In [9]: m.AddBond(5,6,Chem.BondType.ZERO)
Out[9]: 7

In [10]: Chem.SanitizeMol(m)
Out[10]: rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE

In [11]: m.Debug()
Atoms:
	0 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 1 chi: 0
	1 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 1 chi: 0
	2 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 1 chi: 0
	3 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 1 chi: 0
	4 6 C chg: 0  deg: 2 exp: 3 imp: 1 hyb: 3 arom?: 1 chi: 0
	5 7 N chg: 0  deg: 3 exp: 3 imp: 0 hyb: 3 arom?: 1 chi: 0
	6 26 Fe chg: 0  deg: 1 exp: 0 imp: 0 hyb: 4 arom?: 0 chi: 0
Bonds:
	0 0->1 order: 12 conj?: 1 aromatic?: 1
	1 1->2 order: 12 conj?: 1 aromatic?: 1
	2 2->3 order: 12 conj?: 1 aromatic?: 1
	3 3->4 order: 12 conj?: 1 aromatic?: 1
	4 4->5 order: 12 conj?: 1 aromatic?: 1
	5 5->0 order: 12 conj?: 1 aromatic?: 1
	6 5->6 order: 21 conj?: 0 aromatic?: 0

In [12]: m = Chem.MolFromSmiles('c1ccccn1.[Fe]',sanitize=False)

In [13]: m = Chem.RWMol(m)

In [14]: m.AddBond(5,6,Chem.BondType.ZERO)
Out[14]: 7

In [15]: Chem.SanitizeMol(m)
[06:31:34] Can't kekulize mol.  Unkekulized atoms: 0 1 2 3 4

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-15-dd6845fa341c> in <module>()
----> 1 Chem.SanitizeMol(m)

ValueError: Sanitization error: Can't kekulize mol.  Unkekulized atoms: 0 1 2 3 4


```

**Comment by greglandrum** [comment]
Closed by #1714
